src/main/java/org/ow2/proactive/procci/request/InstancesService.java

```diff
@@ -0,0 +1,140 @@
+package org.ow2.proactive.procci.request;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.ow2.proactive.procci.model.cloud.automation.Model;
+import org.ow2.proactive.procci.model.exception.ClientException;
+import org.ow2.proactive.procci.model.occi.infrastructure.Compute;
+import org.ow2.proactive.procci.model.occi.infrastructure.ComputeBuilder;
+import org.ow2.proactive.procci.model.occi.metamodel.Entity;
+import org.ow2.proactive.procci.model.occi.metamodel.Mixin;
+import org.ow2.proactive.procci.model.occi.metamodel.rendering.EntityRendering;
+import org.ow2.proactive.procci.model.utils.ConvertUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.simple.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import static org.ow2.proactive.procci.model.occi.metamodel.constants.Attributes.ID_NAME;
+
+@Component
+public class InstancesService {
```

Comment:
Rename to _InstanceService_?

http://softwareengineering.stackexchange.com/a/103726

Same for other classes using plural in compound  names?